streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -65,7 +65,24 @@
                                    keySchema.hasNextCondition(keyFrom, keyTo, from, to),
                                    binaryFrom, binaryTo);
     }
-
+    
+    @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        
+        final List<Segment> searchSpace = segments.allSegments();
+        
+        return new TimeRangeSegmentIterator(searchSpace.iterator(),
```

Comment:
Rather than adding a subclass of `SegmentIterator` we could use a functional interface to get the next iterator. The `hasNext` method in this is identical to the one in `SegmentIterator` expect for one line.

Reply:
@dguy Yeah, the primary difference between ```SegmentIterator``` and ```TimeRangeSegmentIterator``` is the difference in calling different ```KeyValueIterator``` methods. Namely, ```all()``` and ```range()```. But I don't think the functional interface is needed, because all we have to do is extract the place where ```all()``` and ```range()``` is called into a method (e.g. ```getAll(K from, K to)```).  And inside this new method, check which one to use: ```all()``` or ```range()```. if the keys are null, then we call ```all()``` else we call ```range()```. That might be a better approach. What do you think?

Reply:
sounds good